upp/stages/resampling.py

```diff
@@ -190,9 +216,14 @@ def run(self):
             "[bold green]Checking requested num_jets based on a sampling fraction of"
             f" {self.config.sampling_fraction}..."
         )
-        for c in self.components:
-            sampling_frac = 1 if c.is_target(self.config.target) else self.config.sampling_fraction
-            c.check_num_jets(c.num_jets, sampling_frac=sampling_frac, cuts=c.cuts)
+        if self.config.sampling_fraction == "auto" or self.config.sampling_fraction is None:
+            self.set_auto_sampling_fraction()
+        else:
+            for c in self.components:
+                sampling_frac = (
+                    1 if c.is_target(self.config.target) else self.config.sampling_fraction
+                )
+                c.check_num_jets(c.num_jets, sampling_frac=sampling_frac, cuts=c.cuts)
```

Comment:
the logic of where `check_num_jets` is called is slightly convoluted now. On L183 it is called only for the target component, but I don't see it called anywhere for the non-target components after their optimal sampling fraction has been determined.

Can we factorise the logic a bit better?
```python
# first determine sampling fraction
sampling_frac = 1 if c.is_target(self.config.target) else self.config.sampling_fraction
if self.config.sampling_fraction == "auto" or self.config.sampling_fraction is None:
    self.set_auto_sampling_fraction()
  
# now run the num jets check
for c in self.components:
    c.check_num_jets(c.num_jets, sampling_frac=sampling_frac, cuts=c.cuts)
```

Hopefully that makes sense!

Reply:
I can make it like that. The motivation was that if we use "auto" for the sampling the get_auto_sampling_frac checks if there is enough jets while computing the fraction. I agree that it may be convoluted so I will try to do it the way you say.